DXRando/DeusEx/Classes/PlayerMoverToggleTrigger.uc

```diff
@@ -0,0 +1,64 @@
+class PlayerMoverToggleTrigger extends MoverToggleTrigger;
+
+var Vector loc;
+var float radius;
+
+var bool playerMoved;
+
+static function PlayerMoverToggleTrigger CreatePMTT(
+    Actor a,
+    name flagName,
+    name tag,
+    byte keyFrame1,
+    byte keyFrame2,
+    float duration,
+    Vector loc,
+    optional int flagExpiration,
+    optional float radius
+) {
+    local PlayerMoverToggleTrigger pmtt;
+
+    if (a == None || tag == '' || flagName == '') return None;
+
+    pmtt = a.Spawn(class'PlayerMoverToggleTrigger',, tag);
+    pmtt.InitPMTT(flagName, keyFrame1, keyFrame2, duration, loc, flagExpiration, radius);
+    return pmtt;
+}
+
+function InitPMTT(
+    name flagName,
+    byte keyFrame1,
+    byte keyFrame2,
+    float duration,
+    Vector loc,
+    optional int flagExpiration,
+    optional float radius
+) {
+    InitMTT(flagName, keyFrame1, keyFrame2, duration, flagExpiration);
+
+    self.loc = loc;
+    self.radius = radius;
+}
+
+function PostPostBeginPlay()
+{
+    playerMoved = false;
+    Super.PostPostBeginPlay();
+}
+
+event Tick(float deltaTime)
+{
+    local DeusExPlayer player;
+
+    Super.Tick(deltaTime);
+
+    if (moversMoved && (playerMoved == false)) {
+        player = DeusExPlayer(GetPlayerPawn());
+        if (player == None) return;
+
+        if (player.location == loc || (radius != 0 && VSize(player.location - loc) > radius))
+            playerMoved = true;
```

Comment:
Shouldn't this set playerMoved to true even if they aren't in the radius? Otherwise this seems like it'll catch the player if they come in through the back door and then walk towards the elevator

Reply:
Oh I see, the else is the one that moves, then the else should probably also set playerMoved to true?

Reply:
The move isn't always successful, likely when the elevator isn't in its final position yet (so it throws the player up again). This way it keeps trying until it sees the next tick that it worked.

Reply:
I believe the SetLocation function returns a bool for success, but also maybe checking location on the next frame could go bad if the player moves during the frame? Might depend on the order of the actors in the linked list?

Reply:
SetLocation is returning true even when it doesn't end up working. The player's location really does change, but by the next tick they're in the air again. Moving while entering the map doesn't break anything.